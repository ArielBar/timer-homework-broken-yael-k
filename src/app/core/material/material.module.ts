import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatIconModule } from '@angular/material/icon';
import { MatCardModule } from '@angular/material/card';
import { MatButtonModule } from '@angular/material/button';
import { MatInputModule } from '@angular/material/input';

const modules = [
  CommonModule,
  MatCardModule,
  MatButtonModule,
  MatInputModule,
  MatIconModule,
];
@NgModule({
  imports: [...modules],
  exports: [...modules],
})
export class MaterialModule {}
