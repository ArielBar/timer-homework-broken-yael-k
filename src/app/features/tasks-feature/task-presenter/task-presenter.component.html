<mat-card class="row">
  <p class="input-name">{{ name | uppercase  }}</p>
  <p>{{ timer | async | minuteSeconds }}</p>
  <button mat-fab (click)="click()">
    <mat-icon>{{ buttonText }}</mat-icon>
  </button>
</mat-card>
