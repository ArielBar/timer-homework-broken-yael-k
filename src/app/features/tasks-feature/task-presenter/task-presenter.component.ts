import {
  Component,
  Input,
  Output,
  EventEmitter,
  ChangeDetectionStrategy
} from '@angular/core';
import { TaskModel } from '../../../core/models/task-model';
import { Observable } from 'rxjs';
import { buttonText } from 'src/app/core/models/button-text';

@Component({
  selector: 'app-task-presenter',
  templateUrl: './task-presenter.component.html',
  styleUrls: ['./task-presenter.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class TaskPresenterComponent implements TaskModel {
  
  @Input() id: number;
  @Input() name: string;
  @Input() timer: Observable<number>;
  @Input() buttonText: buttonText;

  @Output() clicked = new EventEmitter<number>();

  public click() {
    this.clicked.emit(this.id);
  }
}
