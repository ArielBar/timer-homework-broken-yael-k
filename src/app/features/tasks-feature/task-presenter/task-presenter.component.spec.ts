import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';

import { TaskPresenterComponent } from './task-presenter.component';
import { Pipe, PipeTransform } from '@angular/core';
import { of } from 'rxjs';
import { MatIconModule } from '@angular/material/icon';
import { MatCardModule } from '@angular/material/card';

describe('TaskPresenterComponent', () => {
  let component: TaskPresenterComponent;
  let fixture: ComponentFixture<TaskPresenterComponent>;
  @Pipe({ name: 'minuteSeconds' })
  class MockPipe implements PipeTransform {
    transform(value: number): number {
      return value;
    }
  }
  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [TaskPresenterComponent, MockPipe],
      imports: [MatIconModule, MatCardModule],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TaskPresenterComponent);
    component = fixture.componentInstance;
    // Arrange
    Object.assign(component, {
      id: 1,
      name: 'some name',
      buttonText: 'pause',
      timer: of(10),
    })

    // act
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it('should render task', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
