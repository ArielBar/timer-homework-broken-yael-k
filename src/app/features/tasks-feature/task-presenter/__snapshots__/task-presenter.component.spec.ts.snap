// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskPresenterComponent should render task 1`] = `
<div
  id="root1"
>
  <mat-card
    class="mat-card mat-focus-indicator row"
  >
    <p
      class="input-name"
    >
      SOME NAME
    </p>
    <p>
      10
    </p>
    <button
      mat-fab=""
    >
      <mat-icon
        aria-hidden="true"
        class="mat-icon notranslate material-icons mat-icon-no-color"
        data-mat-icon-type="font"
        role="img"
      >
        pause
      </mat-icon>
    </button>
  </mat-card>
</div>
`;
